core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -792,6 +793,11 @@ private[spark] object Utils extends Logging {
   lazy val localIpAddress: String = findLocalIpAddress()
   lazy val localIpAddressHostname: String = getAddressHostName(localIpAddress)
 
+  def localIpURI(): String = {
```

Comment:
I have a small concern that this isn't actually returning a URI, but just an IP address in URI-friendly form. Shouldn't callers just use `InetAddresses.toUriString()` if needed? there's already a utility method for this, and doesn't need a new one in `Utils`. Yes, this also ignores the logic behind `localIpAddress` and `localIpAddressHostname`.